src/main/java/duke/data/task/TaskList.java

```diff
@@ -0,0 +1,57 @@
+package duke.data.task;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+public class TaskList {
+
+    private final List<Task> tasks;
+
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Initialises  based on a list of existing tasks.
+     *
+     * @param tasks The lists of tasks
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+    /**
+     * Fetch the task at index taskId
+     *
+     * @param taskId The ID of the task
+     */
+    public Task getTask(int taskId) {
+        return this.tasks.get(taskId - 1);
+    }
+
+    public boolean addTask(Task task) {
+        return this.tasks.add(task);
+    }
+
+    public Task deleteTask(int taskId) {
+        return this.tasks.remove(taskId - 1);
+    }
+
+    public TaskList search(String keyword) {
+        List<Task> filteredList = tasks.stream()
```

Comment:
Using such complex operations make the code extremely long and hard to read. Please abstract it out into different functions or separate out the logic into different lines